nemo/utils/exp_manager.py

```diff
@@ -106,124 +240,199 @@ def exp_manager(trainer: 'pytorch_lightning.Trainer', cfg: Optional[Union[DictCo
             "You are running multi-gpu without ddp.Please note that this is not tested in NeMo and could result in "
             "errors."
         )
+
+
+def check_resume(
+    trainer: 'pytorch_lightning.Trainer',
+    previous_log_dir: str,
+    explicit_log_dir: str,
+    root_dir: str,
+    name: str,
+    version: str,
+    resume_past_end: bool = False,
+) -> (Path, str, str, str):
+    """Checks that resume=True was used correctly with the arguments pass to exp_manager.
+
+    Returns:
+        log_dir (Path): the log_dir
+        root_dir (str): the base root_dir without name nor version
+        name (str): The name of the experiment
+        version (str): The version of the experiment
+
+    Raises:
+        LoggerMisconfigurationError: If trainer is incompatible with arguments
+        ValueError: If resume is True and checkpoints could not be find.
+    """
+    if trainer.logger is not None:
+        raise LoggerMisconfigurationError(
+            "The pytorch lightning trainer that was passed to exp_manager contained a logger and resume was set to "
+            "True. Please remove the logger from the lightning trainer."
+        )
+    if explicit_log_dir or root_dir or name or version:
+        logging.error(
+            f"exp_manager received resume == True, and at least one of explicit_log_dir: {explicit_log_dir}, root_dir: "
+            f"{root_dir}, name: {name}, or version: {version}. Please note that "
+            "explicit_log_dir, root_dir, name, and version will be ignored."
+        )
+    if not previous_log_dir:
+        raise ValueError(f"Resuming requires the previous_log_dir {previous_log_dir} to be passed to exp_manager")
+
+    checkpoint_dir = Path(Path(previous_log_dir) / "checkpoints")
```

Comment:
Here its assumed to be a path. Can we drop this argument entirely, and simply do the following logic

If resume = True, dont add a timestamp and simply create a checkpoints directory under the path that exp_manager normally resolves to. Then on next run check if that path already has checkpoints dir (and a valid ckpt file inside it (len(glob.glob(<path>/*.pt)) > 0) and simply pick the "end" or "last" checkpoints? 

The current argument `previous_log_dir` requires you to explicitly provide a previous path (which you have to manually resolve by yourself using the logic that exp_manager will use to construct paths). It seems highly cumbersome when exp_manager is the one already resolving paths for the user.

Reply:
removed